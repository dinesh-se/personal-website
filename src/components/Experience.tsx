import Image from 'next/image';

import { Experience as ExperienceType } from '@types';
import IconBriefcase from '@root/public/assets/briefcase.svg';

const Experience = ({
  organizations
}: ExperienceType) => {
  const formatDate = (date: string) => new Date(date).toLocaleDateString('en', { year: 'numeric', month: 'short' });

  return (
    <div className="p-6 rounded-2xl border border-zinc-700/40">
      <h4 className="flex font-semibold tracking-wider space-x-4">
        <IconBriefcase className="h-6 w-6 svg-icon" />
        <span>Experience</span> 
      </h4>
      <div className="mt-6 space-y-4">
        {organizations.map(({
          orgName, orgLogo, title, from, to
        }) => (
          <div className="flex w-full space-x-4" key={`${orgName}-${from}-${to}`}>
            <div className="h-10 w-10 relative flex items-center overflow-hidden justify-center bg-white shadow-md border border-slate-300 dark:border-zinc-500 rounded-md"> 
              <Image 
                src={orgLogo.url} 
                alt={`${orgName} Logo`} 
                className="h-auto w-auto object-cover text-transparent" 
                width="40" 
                height="40"
              />
            </div>
            <div className="flex flex-col flex-1">
              <h4 className="font-semibold tracking-wider">{orgName}</h4>
              <div className="flex flex-wrap justify-start sm:justify-between w-full sm:space-x-4 text-sm text-zinc-400">
                <p className="text-sm">{title}</p>
                <div className="text-sm">
                  <time dateTime="2023-05-25T00:00:00.000Z">{formatDate(from)}</time> 
                  <span aria-hidden="true"> â€” </span>
                  { to ? <time dateTime="2024-01-06T00:01:17.965Z">{formatDate(to)}</time> : <span>Present</span>}
                </div>
              </div>
            </div>
          </div>
        ))}
      </div>
      <a 
        className="flex items-center justify-center 
          rounded-md 
          text-sm font-semibold 
          mt-6 py-2 px-3 
          transition 
          bg-gray-300 hover:bg-gray-500 hover:text-gray-100
           dark:bg-zinc-800/50 dark:hover:text-zinc-50 dark:hover:bg-zinc-800
           cursor-pointer 
           active:ring focus:ring" 
        href="https://drive.google.com/file/d/13xI1xgpGBtYOGV9nOr0nplmb88STtI6C/view?usp=sharing">
        Download Resume
      </a> 
    </div>
  )
}

export default Experience;
