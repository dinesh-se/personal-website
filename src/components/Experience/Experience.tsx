import Image from 'next/image';

import IconBriefcase from '@root/public/assets/briefcase.svg';

import { Experience as ExperienceType } from '@types';

const Experience = ({ organizations }: ExperienceType) => {
	const formatDate = (date: string) =>
		new Date(date).toLocaleDateString('en', {
			year: 'numeric',
			month: 'short',
		});

	return (
		<div className="rounded-2xl border border-zinc-700/40 p-6">
			<h4 className="flex space-x-4 font-semibold tracking-wider">
				<IconBriefcase className="svg-icon h-6 w-6" />
				<span>Experience</span>
			</h4>
			<div className="mt-6 space-y-4">
				{organizations.map(({ orgName, orgLogo, title, from, to }) => (
					<div
						className="flex w-full space-x-4"
						key={`${orgName}-${from}-${to}`}
					>
						<div className="relative flex h-10 w-10 items-center justify-center overflow-hidden rounded-md border border-slate-300 bg-white shadow-md dark:border-zinc-500">
							<Image
								src={orgLogo.url}
								alt={`${orgName} Logo`}
								className="h-auto w-auto object-cover text-transparent"
								width="40"
								height="40"
							/>
						</div>
						<div className="flex flex-1 flex-col">
							<h4 className="font-semibold tracking-wider">{orgName}</h4>
							<div className="flex w-full flex-wrap justify-start text-sm text-zinc-400 sm:justify-between sm:space-x-4">
								<p className="text-sm">{title}</p>
								<div className="text-sm">
									<time dateTime="2023-05-25T00:00:00.000Z">
										{formatDate(from)}
									</time>
									<span aria-hidden="true"> â€” </span>
									{to ? (
										<time dateTime="2024-01-06T00:01:17.965Z">
											{formatDate(to)}
										</time>
									) : (
										<span>Present</span>
									)}
								</div>
							</div>
						</div>
					</div>
				))}
			</div>
			<a
				className="mt-6 flex cursor-pointer
          items-center
          justify-center rounded-md
          bg-gray-300 px-3 py-2
          text-sm
          font-semibold transition hover:bg-gray-500
           hover:text-gray-100 focus:ring active:ring
           dark:bg-zinc-800/50
           dark:hover:bg-zinc-800 dark:hover:text-zinc-50"
				href="https://link.dineshharibabu.in/resume"
			>
				Download Resume
			</a>
		</div>
	);
};

export default Experience;
