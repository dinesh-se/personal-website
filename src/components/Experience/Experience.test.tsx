import { render } from '@testing-library/react';

import { Experience as ExperienceType } from '@types';

import Experience from './Experience';

describe('Experience', () => {
	it('renders correctly', () => {
		const props: ExperienceType = {
			organizations: [
				{
					orgName: 'Organization 1',
					orgLogo: {
						url: 'https://test.org/logo-1',
					},
					title: 'Title 1',
					from: '01/01/2024',
					to: '31/01/2024',
				},
				{
					orgName: 'Organization 2',
					orgLogo: {
						url: 'https://test.org/logo-2',
					},
					title: 'Title 2',
					from: '01/02/2024',
					to: '29/02/2024',
				},
			],
		};
		const { container } = render(<Experience {...props} />);
		expect(container).toMatchSnapshot();
	});
});
