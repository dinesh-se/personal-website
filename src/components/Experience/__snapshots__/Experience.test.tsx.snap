// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience renders correctly 1`] = `
<div>
  <div
    class="rounded-2xl border border-zinc-700/40 p-6"
  >
    <h4
      class="flex space-x-4 font-semibold tracking-wider"
    >
      <svg
        class="svg-icon h-6 w-6"
      />
      <span>
        Experience
      </span>
    </h4>
    <div
      class="mt-6 space-y-4"
    >
      <div
        class="flex w-full space-x-4"
      >
        <div
          class="relative flex h-10 w-10 items-center justify-center overflow-hidden rounded-md border border-slate-300 bg-white shadow-md dark:border-zinc-500"
        >
          <img
            alt="Organization 1 Logo"
            class="h-auto w-auto object-cover text-transparent"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Ftest.org%2Flogo-1&w=96&q=75"
            srcset="/_next/image?url=https%3A%2F%2Ftest.org%2Flogo-1&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Ftest.org%2Flogo-1&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="flex flex-1 flex-col"
        >
          <h4
            class="font-semibold tracking-wider"
          >
            Organization 1
          </h4>
          <div
            class="flex w-full flex-wrap justify-start text-sm text-zinc-400 sm:justify-between sm:space-x-4"
          >
            <p
              class="text-sm"
            >
              Title 1
            </p>
            <div
              class="text-sm"
            >
              <time
                datetime="2023-05-25T00:00:00.000Z"
              >
                Jan 2024
              </time>
              <span
                aria-hidden="true"
              >
                 — 
              </span>
              <time
                datetime="2024-01-06T00:01:17.965Z"
              >
                Invalid Date
              </time>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex w-full space-x-4"
      >
        <div
          class="relative flex h-10 w-10 items-center justify-center overflow-hidden rounded-md border border-slate-300 bg-white shadow-md dark:border-zinc-500"
        >
          <img
            alt="Organization 2 Logo"
            class="h-auto w-auto object-cover text-transparent"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Ftest.org%2Flogo-2&w=96&q=75"
            srcset="/_next/image?url=https%3A%2F%2Ftest.org%2Flogo-2&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Ftest.org%2Flogo-2&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <div
          class="flex flex-1 flex-col"
        >
          <h4
            class="font-semibold tracking-wider"
          >
            Organization 2
          </h4>
          <div
            class="flex w-full flex-wrap justify-start text-sm text-zinc-400 sm:justify-between sm:space-x-4"
          >
            <p
              class="text-sm"
            >
              Title 2
            </p>
            <div
              class="text-sm"
            >
              <time
                datetime="2023-05-25T00:00:00.000Z"
              >
                Jan 2024
              </time>
              <span
                aria-hidden="true"
              >
                 — 
              </span>
              <time
                datetime="2024-01-06T00:01:17.965Z"
              >
                Invalid Date
              </time>
            </div>
          </div>
        </div>
      </div>
    </div>
    <a
      class="mt-6 flex cursor-pointer  items-center  justify-center rounded-md  bg-gray-300 px-3 py-2  text-sm  font-semibold transition hover:bg-gray-500 hover:text-gray-100 focus:ring active:ring dark:bg-zinc-800/50  dark:hover:bg-zinc-800 dark:hover:text-zinc-50"
      href="https://drive.google.com/file/d/13xI1xgpGBtYOGV9nOr0nplmb88STtI6C/view?usp=sharing"
    >
      Download Resume
    </a>
  </div>
</div>
`;
